packages/web/components/bridge/immersive/immersive-bridge.tsx

```diff
@@ -0,0 +1,272 @@
+import { Transition } from "@headlessui/react";
+import { MinimalAsset } from "@osmosis-labs/types";
+import { isNil } from "@osmosis-labs/utils";
+import { memo, PropsWithChildren, useState } from "react";
+import { useLockBodyScroll } from "react-use";
+
+import { Icon } from "~/components/assets";
+import { AmountScreen } from "~/components/bridge/immersive/amount-screen";
+import { AssetSelectScreen } from "~/components/bridge/immersive/asset-select-screen";
+import { Screen, ScreenManager } from "~/components/screen-manager";
+import { StepProgress } from "~/components/stepper/progress-bar";
+import { Button, IconButton } from "~/components/ui/button";
+import { EventName } from "~/config";
+import { BridgeFlowProvider } from "~/hooks/bridge";
+import { useAmplitudeAnalytics } from "~/hooks/use-amplitude-analytics";
+import { useDisclosure } from "~/hooks/use-disclosure";
+import { FiatRampKey } from "~/integrations";
+import { ModalCloseButton } from "~/modals";
+import { FiatOnrampSelectionModal } from "~/modals/fiat-on-ramp-selection";
+import { FiatRampsModal } from "~/modals/fiat-ramps";
+import { api } from "~/utils/trpc";
+
+enum ImmersiveBridgeScreens {
+  Asset = "0",
+  Amount = "1",
+  Review = "2",
+}
+
+const MemoizedChildren = memo(({ children }: PropsWithChildren<{}>) => {
+  return <>{children}</>;
+});
+
+export const ImmersiveBridgeFlow = ({
+  Provider,
+  children,
+}: PropsWithChildren<BridgeFlowProvider>) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const [step, setStep] = useState<ImmersiveBridgeScreens>(
+    ImmersiveBridgeScreens.Asset
+  );
+  const [type, setType] = useState<"deposit" | "withdraw">("deposit");
+  const { logEvent } = useAmplitudeAnalytics();
+
+  const apiUtils = api.useUtils();
+
+  const [assetInOsmosis, setAssetInOsmosis] = useState<MinimalAsset>();
```

Comment:
What's the plan for handling multi variant assets? Will this be handled within the amount screen?

Reply:
We may want to pass the Osmosis asset variants as an array here since we do some custom filtering in the `getImmersiveBridgeAssets` tRPC procedure in the component before this.

I'm thinking then, in this component (or in a hook here), we do the logic of aggregating the bridge provider data returned for each Osmosis variant (selectable chains + assets)

Reply:
The thing about variants is that they are not necessarily assets in Osmosis. I don't want to complicate the state flow or cause unnecessary re-renders, as these variants come from providers. However, I agree that we should aggregate this data somewhere. Creating a hook makes sense, and that's my current plan. I'll most likely keep them in a separate state to make it easier to manage

Reply:
By variants I mean the variants on Osmosis. When the user clicks USDC, we're going to pass Noble USDC to each bridge provider `getSupportedAssets`, but we're also going to be passing USDC.axl (and any other variants on Osmosis) to `getSupportedAssets` separately. Then, at the end we try to swap whatever variant the user owns into our canonical version (Noble USDC, an alloy, etc)

Reply:
You're right! I'll fix this